docs/configuration/defaults.md

```diff
@@ -0,0 +1,221 @@
+
+```yaml
+input_features:
+    -   
+        name: title
+        type: text
+        preprocessing:
+            word_length_limit: 20
+    -   
+        name: body
+        type: text
+        preprocessing:
+            word_length_limit: 2000
+defaults:
+    text:
+        preprocessing:
+            most_common_word: 10000
+```
+
+#### Tokenizers
+
+Sequence, text, and set features tokenize features as part of preprocessing. There are several tokenization options that can be specified:
+
+- `characters`: splits every character of the input string in a separate token.
+- `space`: splits on space characters using the regex `\s+`.
+- `space_punct`: splits on space characters and punctuation using the regex `\w+|[^\w\s]`.
+- `underscore`: splits on the underscore character `_`.
+- `comma`: splits on the underscore character `,`.
+- `untokenized`: treats the whole string as a single token.
+- `stripped`: treats the whole string as a single token after removing spaces at the beginning and at the end of the string.
+- `hf_tokenizer`: uses the Hugging Face AutoTokenizer which uses a `pretrained_model_name_or_path` parameter to decide which tokenizer to load.
+- Language specific tokenizers: [spaCy](https://spacy.io) based language tokenizers.
+
+The [spaCy](https://spacy.io) based tokenizers are functions that use the powerful tokenization and NLP preprocessing models provided the library.
+Several languages are available: English (code `en`), Italian (code `it`), Spanish (code `es`), German (code `de`), French (code `fr`), Portuguese (code `pt`), Dutch (code `nl`), Greek (code `el`), Chinese (code `zh`), Danish (code `da`), Dutch (code `el`), Japanese (code `ja`), Lithuanian (code `lt`), Norwegian (code `nb`), Polish (code `pl`), Romanian (code `ro`) and Multi (code `xx`, useful in case you have a dataset containing different languages).
+
+For each language different functions are available:
+
+- `tokenize`: uses spaCy tokenizer,
+- `tokenize_filter`: uses spaCy tokenizer and filters out punctuation, numbers, stopwords and words shorter than 3 characters,
+- `tokenize_remove_stopwords`: uses spaCy tokenizer and filters out stopwords,
+- `lemmatize`: uses spaCy lemmatizer,
+- `lemmatize_filter`: uses spaCy lemmatizer and filters out punctuation, numbers, stopwords and words shorter than 3 characters,
+- `lemmatize_remove_stopwords`: uses spaCy lemmatize and filters out stopwords.
+
+In order to use these options, you must first download the the spaCy model:
+
+```
+python -m spacy download <language_code>
+```
+
+and provide `<language>_<function>` as `tokenizer` like: `english_tokenizer`, `italian_lemmatize_filter`, `multi_tokenize_filter` and so on.
+More details on the models can be found in the [spaCy documentation](https://spacy.io/models).
+
+### Type-Global Encoder
+
+Specify the encoder type and encoder related parameters across all input features of a
+certain data type. For example:
```

Comment:
One thing I would add here is that you are specifying one encoder to be shared across all these parameters (like we discussed on our huddle). Basically making it clear that the parameters in addition to type must correspond to that encoder type since the default is one encoder that will be used for all text features or whatever feature type they choose.

Reply:
I agree! I've made this more clear now